tests/test-rest-users-controller.php

```diff
@@ -322,6 +323,44 @@ public function test_get_items_slug_query() {
 		$this->assertEquals( $id2, $data[0]['id'] );
 	}
 
+	// Note: Do not test using editor role as there is an editor role created in testing and it makes it hard to test this functionality.
+	public function test_get_items_roles() {
+		wp_set_current_user( $this->user );
+		$tango = $this->factory->user->create( array( 'display_name' => 'tango', 'role' => 'subscriber' ) );
+		$yolo  = $this->factory->user->create( array( 'display_name' => 'yolo', 'role' => 'author' ) );
+		$request = new WP_REST_Request( 'GET', '/wp/v2/users' );
+		$request->set_param( 'roles', 'author,subscriber' );
+		$response = $this->server->dispatch( $request );
+		$data = $response->get_data();
+		$this->assertEquals( 2, count( $data ) );
+		$this->assertEquals( $tango, $data[0]['id'] );
+		$this->assertEquals( $yolo, $data[1]['id'] );
+		$request->set_param( 'roles', 'author' );
+		$response = $this->server->dispatch( $request );
+		$data = $response->get_data();
+		$this->assertEquals( 1, count( $data ) );
+		$this->assertEquals( $yolo, $data[0]['id'] );
+		wp_set_current_user( 0 );
+		$request->set_param( 'roles', 'author' );
+		$response = $this->server->dispatch( $request );
+		$this->assertErrorResponse( 'rest_user_cannot_view', $response, 401 );
+		wp_set_current_user( $this->editor );
+		$request->set_param( 'roles', 'author' );
+		$response = $this->server->dispatch( $request );
+		$this->assertErrorResponse( 'rest_user_cannot_view', $response, 403 );
+	}
+
+	public function test_get_items_invalid_roles() {
+		wp_set_current_user( $this->user );
+		$lolz = $this->factory->user->create( array( 'display_name' => 'lolz', 'role' => 'author' ) );
+		$request = new WP_REST_Request( 'GET', '/wp/v2/users' );
+		$request->set_param( 'roles', 'ilovesteak,author' );
+		$response = $this->server->dispatch( $request );
+		$data = $response->get_data();
+		$this->assertEquals( 1, count( $data ) );
```

Comment:
If you are referring to role as a taxonomy for users (When I hear the term taxonomy I think of post taxonomies, so just want to double check.) then I am pretty sure it returns an empty set but would have to double check and will add a test.  What is the desired functionality because it can go either way?